apps/dashboard/src/components/LeftPanel/ElementRow.tsx

```diff
@@ -56,7 +95,35 @@ export function ElementRow({ element }: ElementRowProps) {
           ? <MagicWandIcon height="1.4em" width="1.4em" />
           : null
         }
-        {element.name}
+        {isEditing ? (
+          <form className="flex items-center gap-2" onSubmit={onSubmit} ref={formRef}>
+            <input
+              className="w-48"
+              id="elementNameInput"
+              onChange={({ target }) => {
+                updateElement({
+                  ...element,
+                  name: target.value
+                })
+              }}
+              onKeyDown={(event) => {
+                if (event.key === "Enter" || event.key === "Escape") {
+                  formRef.current?.requestSubmit()
+                }
+              }}
+              type="text"
+              value={element.name}
+            />
+            <button
+              className="text-gray-600 hover:text-gray-900"
+              type="submit"
+            >
+              <CheckIcon />
+            </button>
+          </form>
+        ) : (<>
+          {element.name}
+        </>)}
```

Comment:
I *think* we can refactor this a bit:

```suggestion
        ) : element.name}
```